activerecord/lib/active_record/scoping/named.rb

```diff
@@ -171,6 +171,12 @@ def scope(name, body, &block)
               "a class method with the same name."
           end
 
+          if Relation.instance_methods.include?(name)
```

Comment:
`Relation.instance_methods` also includes all instance methods defined on `Object`. We shouldn't reject those; declaring a scope named `object_id` works just fine, for example, even if it's an odd thing to do.

I think [`method_defined_within?(name, Relation)`](https://github.com/rails/rails/blob/v5.1.4/activerecord/lib/active_record/attribute_methods.rb#L111) will do the right thing here.